app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala

```diff
@@ -18,46 +18,44 @@ package uk.gov.hmrc.customs.notification.services
 
 import javax.inject.{Inject, Singleton}
 
-import play.api.mvc.Headers
-import uk.gov.hmrc.customs.notification.connectors.{NotificationQueueConnector, PublicNotificationServiceConnector}
-import uk.gov.hmrc.customs.notification.domain.PublicNotificationRequest
+import uk.gov.hmrc.customs.notification.connectors.{GoogleAnalyticsSenderConnector, NotificationQueueConnector, PublicNotificationServiceConnector}
+import uk.gov.hmrc.customs.notification.controllers.RequestMetaData
+import uk.gov.hmrc.customs.notification.domain.{DeclarantCallbackData, PublicNotificationRequest}
 import uk.gov.hmrc.customs.notification.logging.NotificationLogger
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.xml.NodeSeq
 
-sealed trait SendNotificationResult
-
-case object NotificationSent extends SendNotificationResult
-case object DeclarantCallbackDataNotFound extends SendNotificationResult
-
 @Singleton
-class CustomsNotificationService @Inject() (logger: NotificationLogger,
-                                            publicNotificationRequestService: PublicNotificationRequestService,
-                                            pushConnector: PublicNotificationServiceConnector,
-                                            queueConnector: NotificationQueueConnector
-                                           ) {
-
-  def sendNotification(xml: NodeSeq, headers: Headers)(implicit hc: HeaderCarrier): Future[SendNotificationResult] = {
-
-    publicNotificationRequestService.createRequest(xml, headers) map {
-      case None =>
-        DeclarantCallbackDataNotFound
-      case Some(request) =>
-        sendAsync(request)
-        NotificationSent
-    }
+class CustomsNotificationService @Inject()(logger: NotificationLogger,
+                                           publicNotificationRequestService: PublicNotificationRequestService,
+                                           pushConnector: PublicNotificationServiceConnector,
+                                           queueConnector: NotificationQueueConnector,
+                                           gaConnector: GoogleAnalyticsSenderConnector
+                                          ) {
+  def handleNotification(xml: NodeSeq, callbackDetails: DeclarantCallbackData, metaData: RequestMetaData)(implicit hc: HeaderCarrier): Future[Unit] = {
+
+    gaConnector.send("notificationRequestReceived", s"[ConversationId=${metaData.conversationId}] A notification received for delivery")
+    pushAnxdThenPassOnToPullIfPushFails(publicNotificationRequestService.createRequest(xml, callbackDetails, metaData))
```

Comment:
typo - no `x` in `And`!!
Also, quite a verbose method name. How about something a little shorter like `pushOrPullIfPushFails`?